### ISSUE TITLE
Possible issue with dicomtocsv --image

### ISSUE THREAD

**Issue by dgobbi** [dgobbi/vtk-dicom#167]
By default, dicomtocsv reports at the series level rather than the image level.  When using the --image option and using key matching (-k key=value) it seems to fail to find matching files if the key is not present for all files in the series (perhaps the key must be present in the first image in the series for the matching to work?)

It is important to check whether dicomfind has similar problems with matching.

**Comment by dgobbi** [comment]
Further investigation shows that the symptoms are not as described above.  I have a series of DICOM files for which some files have NumberOfFrames=14 and others do not have the NumberOfFrames tag.

When using `dicomtocsv --image -k NumberOfFrames=14 /path/to/files`, the matching files are correctly reported.

When using `dicomtocsv --image -k NumberOfFrames /path/to/files` the NumberOfFrames is shown to be absent/empty for all images.

However, if I sort the files by InstanceNumber and delete all files before the ones that have NumberOfFrames, then NumberOfFrames is correctly reported for all files.

In conclusion: the problem seems to be related to whether the first file in the series contains the tag that is being queried.

**Comment by dgobbi** [comment]
The problem also occurs with the vtkDICOMDirectory class that is used with dicomtocsv.  When calling the GetMetaDataForSeries() method, the returned metadata does not include any queried tags that are missing from the first file in the series.

The next step is to check whether this is also true for vtkDICOMParser itself when used with a query.

**Comment by dgobbi** [comment]
Fixed by ffe4293
